native-jfr-demo/README.md

```diff
@@ -0,0 +1,80 @@
+# Recording JFR events with Native Image
+
+This demo shows how to create a custom JDK Flight Recorder (JFR) event and use that in a native executable.
+JFR is a tool to collect diagnostic and profiling data in a running Java application, built into a JVM. 
+GraalVM Native Image supports JFR events and users can use the [`jdk.jfr.Event`](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jfr/jdk/jfr/Event.html) API with a similar experience to using JFR in the JVM. For example, to emit standard JFR events for the `jwebserver` application, you would run:
+
+```bash
+$JAVA_HOME/bin/native-image --enable-monitoring=jfr -m jdk.httpserver
+```
+```bash
+./jdk.httpserver -XX:+FlightRecorder -XX:StartFlightRecording="duration=10s,filename=recording.jfr"
+```
+
+Follow the steps below to record JFR events when running a native executable, enable JFR support, and JFR recording.
+
+> Note: JFR event recording is not yet supported on GraalVM JDK for Windows.
+
+## Preparation
+
+1. Download and  install the latest GraalVM JDK with Native Image using [GraalVM JDK Dowloader](https://github.com/graalvm/graalvm-jdk-downloader):
+    ```bash
+    bash <(curl -sL https://get.graalvm.org/jdk) -c 'native-image,visualvm'
+    ```
+
+2. Install VisualVM by running:
+    ```bash
+    gu install visualvm
+    ``` 
+
+3. Download the demos repository or clone it as follows:
+    ```
+    git clone https://github.com/graalvm/graalvm-demos
+    ```
+
+## Build and Run Native Executables with JFR
+
+1. Change directory to the demo subdirectory: _native-jfr-demo_:
+    ```
+    cd graalvm-demos/native-jfr-demo
+    ```
+2. Compile the Java file using the GraalVM JDK:
+    ```bash
+    $JAVA_HOME/bin/javac JFRDemo.java
+    ```
+    It creates two class files: _JFRDemo$HelloWorldEvent.class_ and _JFRDemo.class_.
+
+3. Build a native executable with VM inspection enabled:
+    ```bash
+    $JAVA_HOME/bin/native-image --enable-monitoring=jfr JFRDemo
+    ```
+    The `--enable-monitoring=jfr` option enables features such as JFR that can be used to inspect the VM.
+
+4. Run the executable and start recording:
+    ```bash
+    ./jfrdemo -XX:StartFlightRecording="filename=recording.jfr"
```

Comment:
You left `-XX:+FlightRecorder` in the `jwebserver` example, so let's use it consistently (it's not needed but also doesn't harm):
```suggestion
    ./jfrdemo -XX:+FlightRecorder -XX:StartFlightRecording="filename=recording.jfr"
```